### ISSUE TITLE
[a11y] Spinner is not read by screen readers

### ISSUE THREAD

**Issue by dpersing** [Shopify/polaris-react#791]
## Issue summary

The [Spinner](https://polaris.shopify.com/components/feedback-indicators/spinner#navigation) has some implementation issues which may cause issues for screen reader users. The `<svg>` element is skipped by VoiceOver on iOS and macOS and NVDA with or without the `aria-label` attribute. There are a couple things that contribute to this:
- There is a bug with how `<svg aria-label="...">` is identified by VO. If the `<svg>` is a child of a focusable element, it will read. Otherwise, it won't. Not even `role="img"` can save it, it seems. (shrug emoji)
- `role="status"` functions as a live region, which requires HTML content to be dynamically updated. Since the `<svg>` has no content (and content inside SVGs is not conveyed consistently anyway), no update can be conveyed.

[There's a bit more info on SVG weirdness here](https://simplyaccessible.com/article/7-solutions-svgs/), and [an MDN entry on `status` that might help](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_status_role). (Full disclosure, I tested and wrote most of the findings for the SVG article. 😇)

#### Spinner output with `accessibilityLabel` value added (HTML)

```
<svg viewBox="0 0 44 44" class="Polaris-Spinner Polaris-Spinner--colorTeal Polaris-Spinner--sizeLarge" aria-label="Please wait" role="status">
<!-- ... -->
</svg>
```

## Steps to reproduce

1. Navigate to [Spinner](https://polaris.shopify.com/components/feedback-indicators/spinner#navigation) using Safari with VoiceOver or NVDA with Firefox.
1. Use the screen reader to read the spinner.
1. Give the spinner an accessible label (name) and reread it.

### Expected

- The spinner will be conveyed programmatically.

### Actual

- The spinner is not conveyed programmatically to screen readers.

## Recommendations

Since SVGs are conveyed inconsistently, I'd recommend hiding the SVG itself and using visually hidden text to convey its purpose instead.

- Use `aria-hidden="true"` to hide the SVG itself and remove the `aria-label`. (Add `focusable="false"` just in case the SVG is implemented inside a focusable element/area to avoid bugs in some browsers.)
- Create a visually hidden `<span>` or `<div>` to serve as the text container.

Additional steps will depend on how the spinner should be conveyed, which may depend on the context. This might be a good opportunity for documentation instead of implementation.

1. To make the spinner read dynamically when it appears, without moving focus, additionally use `role="status"`:
  - Load the `<span role="status" class="Polaris-VisuallyHidden">` in the DOM by default.
  - Append/update the `accessibleLabel` when the spinner is displayed. This will trigger reading dynamically.
2. Alternately, if the value doesn't need to be read dynamically and should just be accessed inline, or focus is sent to it for reading, the text can just be visually hidden and doesn't need to be dynamically added.

### Recommended spinner output with `accessibleLabel` valued added, live region version (HTML)

```
<svg aria-hidden="true" focusable="false" viewBox="0 0 44 44" class="Polaris-Spinner Polaris-Spinner--colorTeal Polaris-Spinner--sizeLarge"><!-- ... --></svg>
<span role="status" class="Polaris-VisuallyHidden">Please wait</span>
```

### Recommended spinner output with `accessibleLabel` valued added, "static" or focused version (HTML)

```
<div><!-- send focus to me, for example -->
<svg aria-hidden="true" focusable="false" viewBox="0 0 44 44" class="Polaris-Spinner Polaris-Spinner--colorTeal Polaris-Spinner--sizeLarge"><!-- ... --></svg>
<span class="Polaris-VisuallyHidden">Please wait</span>
</div>
```

## Specifications

- **Component:** Spinner
- **WCAG SCs:** [4.1.3 Status Messages (Level AA)](https://www.w3.org/TR/WCAG21/#status-messages)
- **Testing done with:**
    - VoiceOver/Safari/macOS
    - VoiceOver/Safari/iOS
    - NVDA/Firefox/Windows 10

**Comment by dpersing** [comment]
Adding to list filters since this component is relied on as the page updates underneath the sheet. Since nothing is announced, screen reader users may be confused about why the screen appears to be frozen.
